doc/pairs.md

```diff
@@ -0,0 +1,35 @@
+# Pairs examples
+
+## Lua Fun
+
+Queries using pairs support the [Lua Fun](https://github.com/luafun/luafun) library. Some examples of working with basic functional functions below.
```

Comment:
I'm not sure that `Queries support LuaFun` is correct. "pairs" is "luafun"-compatible or something like that.
Iterator can't support library - it's not clear.